build-scripts/build_cpp_tests.sh

```diff
@@ -1,15 +1,22 @@
 #!/bin/bash
 
+# Any command line arguments are assumed to be CMake arguments
+CMAKE_ARGS=$@
+
 CMAKE=$(which cmake)
 
 cd ./tests/cpp/
 
 # setup build dirs
+if [[ -d "./build" ]]; then
+    echo "Removing previous cpp unit test build directory"
+    rm -rf ./build
+fi
```

Comment:
Thank you for doing this -- I actually just put in an issue about the need for this!